app/Providers/RouteServiceProvider.php

```diff
@@ -29,6 +29,12 @@ public function boot()
         Route::bind('reciter', function ($value) {
             return Reciter::where('id', $value)->orWhere('slug', $value)->first();
         });
+
+        Route::bind('album', function ($value, $route) {
+            if ($reciter = $route->parameter('reciter')) {
+                return $reciter->albums()->where('year', $value)->firstOrFail();
+            }
```

Comment:
This function doesn't return anything explicitly when the `if` statement resolves to false. Can you add 
 `throw new \Illuminate\Database\Eloquent\ModelNotFoundException();` (you should import that long class name by adding a `use` statement at the top of the file.)